### PR TITLE
[BCI-3862][chainlink] - Change DSL Block primitive to string instead of int

### PR DESCRIPTION
**Pull request by Farber98**
### Task Description:
[Change ChainReader block DSL](https://github.com/smartcontractkit/chainlink-common/blob/4ed45812784df8f781beab8748a5a1b504527990/pkg/loop/internal/pb/chain_reader.proto#L154) uint64 type to string and update EVM implementation to reflect changes

### This PR:
This PR fixes all the reference failures within `chainlink` after modifying `chainlink-common` [block proto field](https://github.com/smartcontractkit/chainlink-common/pull/683/files#diff-9bc4bf4d4deb80e52f62b6362067fc765874dca18a45f5597a18ebdae202de85L154)

### Ticket:
- [BCI-3862](https://smartcontract-it.atlassian.net/browse/BCI-3862)

### Depends On:
- https://github.com/smartcontractkit/chainlink-common/pull/688

[BCI-3862]: https://smartcontract-it.atlassian.net/browse/BCI-3862?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Farber98** [comment]
@ilija42 Once https://github.com/smartcontractkit/chainlink-common/pull/683 gets merged, do I need a separate PR to bump common [like this one](https://github.com/smartcontractkit/chainlink/pull/13623)? Or can I do it in this same PR.

Also, when bumping the ref, do we use UTC for the new timestamp?

Solution:  run `go get github.com/smartcontractkit/chainlink-common@xyz` where `xyz` is the latest commit hash here https://github.com/smartcontractkit/chainlink-common/pull/683/commits. Need to do this in `root`, `/integration-tests`, `/integration-tests/load` and `/core/scripts`. Then run `make gomodtidy` in root

**Review by ilija42 — CHANGES_REQUESTED** [review]
You should first reference the common changes so that CI passes, then once common is merged you can update the reference to the squashed commit hash from common master branch, you don't need separate PRs

File: go.mod [hunk 1]
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/NethermindEth/juno v0.3.1
 	github.com/NethermindEth/starknet.go v0.7.1-0.20240401080518-34a506f3cfdb
 	github.com/XSAM/otelsql v0.27.0
-	github.com/avast/retry-go/v4 v4.6.0
+	github.com/avast/retry-go/v4 v4.5.1
```
Comment (ilija42) [inline go.mod]: why did this downgrade? this is causing build failure

Comment (ilija42) [reply go.mod]: you should revert this change and run tidy

Comment (Farber98) [reply go.mod]: fixed

**Review by ilija42 — CHANGES_REQUESTED** [review]
Looks good, but has merge conflicts

**Review by ilija42 — DISMISSED** [review]

**Review by EasterTheBunny — DISMISSED** [review]

**Review by ilija42 — DISMISSED** [review]

**Review by EasterTheBunny — DISMISSED** [review]

**Review by ilija42 — APPROVED** [review]

**Review by silaslenihan — APPROVED** [review]
LGTM

**Pull request merged**